src/Migrations/PregWithMatchesMigration.hack

```diff
@@ -0,0 +1,96 @@
+/*
+ *  Copyright (c) 2017-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the MIT license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ */
+
+namespace Facebook\HHAST;
+
+use namespace HH\Lib\C;
+
+/**
+ * Migrates preg_match() and preg_match_all() calls with a by-ref $matches
+ * argument to preg_match_with_matches() and preg_match_all_with_matches().
+ */
+final class PregWithMatchesMigration extends BaseMigration {
+
+  const dict<string, string> REPLACEMENTS = dict[
+    'preg_match' => 'preg_match_with_matches',
+    'preg_match_all' => 'preg_match_all_with_matches',
+  ];
+
+  <<__Override>>
+  public function migrateFile(string $_path, Script $root): Script {
+    foreach (
+      $root->getDescendantsOfType(FunctionCallExpression::class) as $node
+    ) {
+      // Only replace calls to functions from the root namespace.
+      $receiver = $node->getReceiver();
+      if (!$receiver is QualifiedName) {
+        continue;
```

Comment:
If you're in the root namespace, or you've done a `use function`, an unqualified name can be a call to the function in the root namespace

A `resolve_function()` probably needs adding, similar to `Facebook\HHAST\resolve_type()`

Reply:
Oh nice, I should be able to reuse that almost without changes.

I thought it would be more complicated so I only handled the common case, but with these it shouldn't be too hard to handle all the cases.